### ISSUE TITLE
Default reading direction for a library

### ISSUE THREAD

**Issue by darthmooguy** [gotson/komga#873]
### Describe your suggested feature

I have all my manga in a dedicated library. It would be great if there was a setting for the library that would put a default reading direction for all the series in the library (kind of like the reading direction for a series being applied to each book in it). It would avoid the need to set the reading direction to right-to-left manually for each new series.

### Other details

Maybe this is more of a concern for komf or other metadata tools, but it’d be useful to have in Komga for those who don’t use them.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to the latest version.
- [X] I will fill out all of the requested information in this form.

**Comment by darthmooguy** [comment]
This was just added in komf (https://github.com/Snd-R/komf/releases/tag/v0.2.0), so in my case I'm happy to just use that. I'll still leave the issue open just in case, but feel free to close it if you don't see any further value to have the feature in Komga!

**Comment by nattycleopatra** [comment]
I don't really agree with @gotson's conclusion in #115. Moving reading direction from Book to Series level was an improvement, but it's still a lot of work for the user having to manually set Series level reading direction for all series added to a library.

I would imagine it to be a fairly common use case to use separate libraries for comics with different reading directions, for instance one library for European comics and another library for Japanese comics. Haven't looked into the relevant parts of the codebase yet, so no idea how much work this would be to implement.

The komf software linked above seems to (currently?) only support manga metadata providers, so for me at least it is not very interesting. It also seems overkill having to run an external service to set library level reading direction.

**Comment by gotson** [comment]
> Haven't looked into the relevant parts of the codebase yet, so no idea how much work this would be to implement.

It's mostly on how to use that information more than adding a field in the library entity. Could be used in 2 ways:
1. if the library has a default, when adding a new series then the default reading direction of the series is set to the one of the library. It would not handle series that are already created and don't have a reading direction.
2. in the webreader, check if the series has a reading direction, and if not fallback to the library's reading direction.

2 works only in the webreader, and doesn't actually populate the series metadata.
1 is cleaner in terms of metadata, but cannot fill missing reading direction for existing series, however that's easy to mass-edit.

I would prefer implementing 1.

**Comment by nattycleopatra** [comment]
Yes, option 1 definitely sounds like the better approach. I very much agree that this should be reflected in the metadata, and it makes sense to apply a library level setting to new series on add.

With regards to the web interface, one could perhaps have one's cake and eat it too in this scenario if changing the library level reading direction came with a prompt of whether to apply it to existing series, and if yes then do GET on `/api/v1/series?library_id={libraryId}` and PATCH the `readingDirection` for each `/api/v1/series/{seriesId}/metadata`. I think that would be nice, but don't know if there are any practical issues with that approach. 

Setting the reading direction for all existing entities via the series metadata endpoint as a one-time operation would at least be easy as you say, whether integrated in the webui or not.

**Comment by gotson** [comment]
Problem with that is you don't know which series should be changed, ideally you only want to change the ones without a reading direction already set.

Doing via the api would be prone to slowness, some people have thousands of series in their libraries, à thousand calls is not practical.

**Comment by gotson** [comment]
It's probably easier to handle the change server side, and always apply. If you're choosing a default reading direction, there's a very big chance you want to apply that on series without a reading direction.

**Comment by darthmooguy** [comment]
To push that idea further, I think the reading direction should be applied to all the series where the reading direction is not locked. For example, if I chose webtoon by mistake, when I change it to ex. right to left, it shouldn't leave everything at webtoon!

**Comment by gotson** [comment]
Debatable, as metadata providers could have set the reading direction but those are never locked. Lock happens only on manual update.

That would effectively override reading direction set via metadata, which would never be reloaded, unless the file changed or from a manual refresh.

**Comment by darthmooguy** [comment]
Aaah, true, I haven't thought about the reading direction from metadata. In that case, in case of a mistake it could be batch changed from the webui, for the affected series. Or maybe just have a prompt/checkbox so the user can choose wether to apply the reading direction to existing series or not? (Only for those without a reading direction set)

**Comment by gotson** [comment]
> Or maybe just have a prompt/checkbox so the user can choose wether to apply the reading direction to existing series or not? (Only for those without a reading direction set)

that's an extra knob that won't be used very often IMHO.

**Comment by eswiig** [comment]
I don't use komf, it seems complicated and unnecessary to set up.

Is there a way to optionally force webtoon reading method?

I only use webtoon reading method regardless of media

but it's really annoying to have to change every time I change chapter. 
I wish there was an option to optionally lock the reading method.

or how can I manipulate the files to be read as webtoon? is there some kind of meta data?

**Comment by gotson** [comment]
> even though it would be nice to OPTIONALLY lock it.

what does that mean?

**Comment by eswiig** [comment]
ah it would mean it would be optional to lock a reading direction server side instead from meta data.
It's a bit tedious having to update metadata / setting on every manga added to komga, but I guess it works for now.

not everyone is tech savvy enough to get komf up and running.

**Comment by gotson** [comment]
> ah it would mean it would be optional to lock a reading direction server side instead from meta data.

lock is optional, just click on the icon. See https://komga.org/guides/edit-metadata.html#locking-and-unlocking
